app/database/db.go

```diff
@@ -115,3 +106,24 @@ func (db *DB) insert(tableName string, data interface{}) error {
 	db = &DB{db.Exec(query, values...)}
 	return db.Error
 }
+
+func decodeValue(value interface{}) (unWrappedValue interface{}, isNull bool, skip bool) {
```

Comment:
Why do you need to do this instead of implementing `Scan` functions on the type?

Reply:
To manage the set and null property. we should figure out by reading the function. Actually, the code was already there, it was just moved to a function as a linter was complaining.

Reply:
But why do you need that instead of managing those properties in `Scan()`?

Reply:
Ah. I think there was something which not working with Scan in this case. But should check if it is still relevant.